### PR TITLE
Fixes to re-enable command driver

### PR DESCRIPTION
**Pull request by simongdavies**
Re-added support for CLI drivers @carolynvs this is the change we discussed earlier this week , can you review please?

### CODE REVIEW

File: pkg/driver/lookup.go [hunk 1]
```diff
@@ -13,9 +12,12 @@ func Lookup(name string) (driver.Driver, error) {
...
-	case "command":
-		return &CommandDriver{Name: name}, nil
 	default:
-		return nil, fmt.Errorf("unsupported driver: %s", name)
+		cmddriver := &CommandDriver{Name: name}
+		if cmddriver.CheckDriverExists() {
+			return cmddriver, nil
+		}
+
+		return nil, fmt.Errorf("unsupported driver or driver not found: %s", name)
```
Comment (carolynvs) [inline pkg/driver/lookup.go]: I recommend adding a mention that the driver is expected to be present on the PATH. Perhaps "unsupported driver or driver not found on the PATH"

File: pkg/driver/command_driver_test.go [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+	"github.com/deislabs/cnab-go/driver"
+)
 
 var _ driver.Driver = &CommandDriver{}
+
+func TestCheckDriverExists(t *testing.T) {
+	name := "missing-driver"
+	cmddriver := &CommandDriver{Name: name}
+	if cmddriver.CheckDriverExists() {
+		t.Errorf("Expected driver %s not to exist", name)
```
Comment (carolynvs) [inline pkg/driver/command_driver_test.go]: You should use `t.Fatalf` here to stop the test. Errorf will report and keep running the test.

**Comment by carolynvs** [comment]
Not sure how to get the CI tests to run when the contributor isn't a member of the org and the CI is managed by brigade, @vdice? Do I have to kick it off through the brig cli instead of through GH?

I ran them locally to confirm that they are passing, which is super official. 😀 

```console
$ go test ./...
ok  	github.com/deislabs/duffle/cmd/duffle	1.095s
ok  	github.com/deislabs/duffle/pkg/builder	0.038s
ok  	github.com/deislabs/duffle/pkg/crypto/digest	0.015s
ok  	github.com/deislabs/duffle/pkg/driver	0.103s
?   	github.com/deislabs/duffle/pkg/duffle	[no test files]
ok  	github.com/deislabs/duffle/pkg/duffle/home	0.067s
ok  	github.com/deislabs/duffle/pkg/duffle/manifest	0.048s
?   	github.com/deislabs/duffle/pkg/imagebuilder	[no test files]
ok  	github.com/deislabs/duffle/pkg/imagebuilder/docker	0.083s
ok  	github.com/deislabs/duffle/pkg/imagebuilder/mock	0.034s
ok  	github.com/deislabs/duffle/pkg/io	0.039s
ok  	github.com/deislabs/duffle/pkg/loader	0.037s
?   	github.com/deislabs/duffle/pkg/ohai	[no test files]
ok  	github.com/deislabs/duffle/pkg/osutil	0.017s
ok  	github.com/deislabs/duffle/pkg/packager	0.071s
ok  	github.com/deislabs/duffle/pkg/reference	0.023s
ok  	github.com/deislabs/duffle/pkg/repo	0.034s
?   	github.com/deislabs/duffle/pkg/repo/remote	[no test files]
?   	github.com/deislabs/duffle/pkg/version	[no test files]
$ echo $?
0
```

**Comment by vdice** [comment]
> Not sure how to get the CI tests to run when the contributor isn't a member of the org and the CI is managed by brigade, @vdice?

We don't currently have a way to do so, but we will soon.  (The feature to enable triggering CI from issue comments - regardless of PR author - is currently in review; expecting it will make it into Brigade's next 1.1.0 release.  I'll certainly help update duffle's CI to enable this when ready.)

For now, indeed, the `super official` 😉 method above will be necessary for the time being.

File: pkg/driver/command_driver_windows.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"os"
+	"strings"
+)
+
+// CheckDriverExists checks to see if the named driver exists
+func (d *CommandDriver) CheckDriverExists() bool {
+
+	path := os.Getenv("PATH")
+	extensions := [3]string{"exe", "cmd", "bat"}
+	for _, dir := range strings.Split(path, ";") {
```
[the remark this answers is not in the document]
Comment (simongdavies) [reply pkg/driver/command_driver_windows.go]: Oh, I misread the docs on where - updated.

**Comment by carolynvs** [comment]
cc @jeremyrickard for another set of 👀

File: pkg/driver/command_driver_test.go [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+	}
+
+	name = "existing-driver"
+	pathSeperator := ":"
```
Comment (glyn) [inline pkg/driver/command_driver_test.go]: Why not use `os.PathListSeparator`?

Comment (glyn) [inline pkg/driver/command_driver_test.go]: Nit: typo in variable name.

File: pkg/driver/command_driver_test.go [hunk 5]
```diff
@@ -1,5 +1,51 @@
...
+	defer os.RemoveAll(dirname)
+	filename := fmt.Sprintf("%s/duffle-%s", dirname, name)
+	newfile, err := os.Create(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newfile.Chmod(0755)
+	path := os.Getenv("PATH")
+	newpath := fmt.Sprintf("%s%s%s", dirname, pathSeperator, path)
```
[the remark this answers is not in the document]
Comment (simongdavies) [reply pkg/driver/command_driver_test.go]: Fileptath join does not make sense here as I am creating a path list not a path but I changed it to use strings.join instead

**Comment by carolynvs** [comment]
Yay! 🎉

**Pull request merged**